packages/shared/src/renderer.tsx

```diff
@@ -461,6 +479,7 @@ export function renderToReact(
     "renderers" | "elementRenderers" | "processMediaReferences"
   >,
 ): RenderToReactResult {
+  const reactElementRenderers = new ReactElementRenderers();
```

Comment:
since this is instantiated per function call, we can simplify it by removing the `randomKey` and using simple `key++` counter in `ReactElementRenderers` class.